### PR TITLE
Makes the 'runtime' tests in mono/tests run on Cygwin with MSVC built mono executable

### PR DESCRIPTION
**Pull request by ntherning**
Makes the 'runtime' tests in mono/tests run on Cygwin with MSVC built mono executable.

Changes:
* Modifies the mono/tests/Makefile.am to pass along the value set in the
  MONO_EXECUTABLE env variable to test-runner.exe using the --runtime command
  line parameter. Defaults to 'mono' which is also the defult in test-
  runner.cs if no --runtime argument is given.
* Updates scripts/ci/run-jenkins.sh to set PATH properly to the dir containing
  the MSVC built libtest.dll. Without this the P/Invoke 'runtime' tests cannot
  locate libtest.dll.

Note: Some tests still fail but at least the test-runner.exe now runs the
tests as expected. The failing tests are subject for future PRs.

### CODE REVIEW

**Comment by kumpera** [comment]
whitelist

File: scripts/ci/run-jenkins.sh [hunk 1]
```diff
@@ -6,8 +6,14 @@ export TEST_HARNESS_VERBOSE=1
...
 if [[ ${label} == 'osx-amd64' ]]; then EXTRA_CONF_FLAGS="--with-libgdiplus=/Library/Frameworks/Mono.framework/Versions/Current/lib/libgdiplus.dylib --enable-nls=no"; fi
-if [[ ${label} == 'w32' ]]; then PLATFORM=Win32; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\Win32\\\bin\\\Release_SGen\\\mono-sgen.exe`";fi
-if [[ ${label} == 'w64' ]]; then PLATFORM=x64; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\x64\\\bin\\\Release_SGen\\\mono-sgen.exe`"; fi
+if [[ ${label} == 'w32' ]]; then PLATFORM=Win32; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; fi
+if [[ ${label} == 'w64' ]]; then PLATFORM=x64; EXTRA_CONF_FLAGS="--host=i686-pc-mingw32"; fi
+
+if [[ ${label} == w* ]]; then
+	export MONO_EXECUTABLE="`cygpath -u ${WORKSPACE}\\\msvc\\\\${PLATFORM}\\\bin\\\Release_SGen\\\mono-sgen.exe`"
+	# without this dir in PATH the P/Invoke tests in mono/tests/ won't find the MSVC built libtest.dll
+	export PATH="$PATH:`cygpath -u ${WORKSPACE}\\\msvc\\\\${PLATFORM}\\\bin\\\Release`"
```
Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: I think a cleaner way would be to add a [dllmap](http://www.mono-project.com/docs/advanced/pinvoke/dllmap/) entry to [tests-config.in](https://github.com/mono/mono/blob/master/mono/tests/tests-config.in) for libtest.dll on Windows that points to the MSVC built library.

I'm not 100% sure whether this works or breaks when running against the cygwin mono where the library is in the same folder (i.e. does it first look in the dllmap location and fail if not found, or local folder first?), but that should be easy to try.

Comment (ntherning) [inline scripts/ci/run-jenkins.sh]: Does indeed sound cleaner. But IIUC the make targets that run when `run-jenkins.sh` invokes `test-wrench` never use `tests-config`. Only seems to be used by the `test-driver` perl script which is never invoked. I tried manually adding a 

```
<dllmap dll="libtest" target="C:\Users\Niklas\Desktop\mono\msvc\x64\bin\Release\libtest.dll" />
```

line to `test-config` but it didn't change anything.

Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: Hmm, indeed. We should get a `--config` option added to the managed test-runner.cs then which passes this along.

(You can try if it works at all by copying the config file as `<appname>.exe.config` next to the exe, then it should get automatically used)

Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: Or use the MONO_CONFIG env var which does the same thing, just for the tests that need it.

**Pull request closed without merging**